frontend/src/host_orchestrator/orchestrator/execcvdcommandaction.go

```diff
@@ -0,0 +1,114 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package orchestrator
+
+import (
+	"log"
+	"os"
+	"path/filepath"
+
+	"github.com/google/android-cuttlefish/frontend/src/host_orchestrator/orchestrator/cvd"
+	apiv1 "github.com/google/android-cuttlefish/frontend/src/liboperator/api/v1"
+	"github.com/google/android-cuttlefish/frontend/src/liboperator/operator"
+)
+
+type ExecCVDCommandActionOpts struct {
+	Request          *apiv1.ExecCVDCommandRequest
+	Paths            IMPaths
+	OperationManager OperationManager
+	ExecContext      ExecContext
+	CVDToolsVersion  AndroidBuild
+	CVDDownloader    CVDDownloader
+	CVDUser          string
+}
+
+type ExecCVDCommandAction struct {
+	req             *apiv1.ExecCVDCommandRequest
+	paths           IMPaths
+	om              OperationManager
+	cvdToolsVersion AndroidBuild
+	cvdDownloader   CVDDownloader
+	execContext     cvd.CVDExecContext
+}
+
+func NewExecCVDCommandAction(opts ExecCVDCommandActionOpts) *ExecCVDCommandAction {
+	return &ExecCVDCommandAction{
+		req:             opts.Request,
+		paths:           opts.Paths,
+		om:              opts.OperationManager,
+		cvdToolsVersion: opts.CVDToolsVersion,
+		cvdDownloader:   opts.CVDDownloader,
+		execContext:     newCVDExecContext(opts.ExecContext, opts.CVDUser),
+	}
+}
+
+func (a *ExecCVDCommandAction) Run() (apiv1.Operation, error) {
+	if err := validateExecCVDCommandRequest(a.req); err != nil {
+		return apiv1.Operation{}, operator.NewBadRequestError("invalid ExecCVDCommandRequest", err)
+	}
+	op := a.om.New()
+	go func(op apiv1.Operation) {
```

Comment:
The response from this api is not an operation.

Reply:
simplified the PR so it executes cvd stop only, this is no longer relevant.